File: app/src/main/java/com/example/capston/pcjari/Address/AddressAdapter.java
package com.example.capston.pcjari.Address;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.example.capston.pcjari.R;

import java.util.ArrayList;

/**
 * Created by KangSeungho on 2017-11-29.
 */

public class AddressAdapter extends BaseAdapter {
    ArrayList<String> addressItems;
    ViewHolder viewHolder;

    @Override
    public int getCount() {
        return addressItems.size();
    }

    @Override
    public String getItem(int position) {
        return addressItems.get(position);
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        Context context = parent.getContext();

        if (convertView == null) {
            LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            convertView = inflater.inflate(R.layout.address_list_view, parent, false);

            viewHolder = new ViewHolder();
            viewHolder.address = (TextView) convertView.findViewById(R.id.address_);

            convertView.setTag(viewHolder);
        }
        else {
            viewHolder = (ViewHolder) convertView.getTag();
        }

        String addressItem = getItem(position);
        viewHolder.address.setText(addressItem);

        return convertView;
    }

    public void addItem(ArrayList<String> addressItems) {
        this.addressItems = addressItems;
    }

    class ViewHolder {
        TextView address;
    }
}

File: app/src/main/java/com/example/capston/pcjari/Address/AddressAdapter.kt
package com.example.capston.pcjari.Address

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import com.example.capston.pcjari.R
import kotlin.collections.ArrayList

/**
 * Created by KangSeungho on 2017-11-29.
 */
class AddressAdapter(var mContext:Context) : BaseAdapter() {
    lateinit var addressItems: ArrayList<String>

    override fun getCount(): Int {
        return addressItems.size
    }

    override fun getItem(position: Int): String {
        return addressItems[position]
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        var view = convertView
        val holder:ViewHolder
        val addressItem = getItem(position)

        if (view == null) {
            val inflater = mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            view = inflater.inflate(R.layout.address_list_view, parent, false) as View

            holder = ViewHolder()
            holder.address = view.findViewById<TextView>(R.id.address_)

            view.tag = holder
        } else {
            holder = view.tag as ViewHolder
        }

        holder.address?.text = addressItem
        return view
    }

    fun addItem(addressItems: ArrayList<String>) {
        this.addressItems = addressItems
    }

    inner class ViewHolder {
        var address: TextView? = null
    }
}